tests/test_ipmitool.py

```diff
@@ -0,0 +1,213 @@
+        {"name": "Negative Hysteresis", "value": "120.000"},
+        {"name": "Minimum sensor range", "value": "Unspecified"},
+        {"name": "Maximum sensor range", "value": "Unspecified"},
+        {"name": "Event Message Control", "value": "Per-threshold"},
+        {"name": "Readable Thresholds", "value": "lcr lnc"},
+        {"name": "Settable Thresholds", "value": ""},
+        {"name": "Threshold Read Mask", "value": "lcr lnc"},
+        {"name": "Assertion Events", "value": ""},
+        {"name": "Assertions Enabled", "value": "lnc- lcr-"},
+        {"name": "Deassertions Enabled", "value": "lnc- lcr-"},
+    ],
+}]
+
+ipmitool_lan_print = """
+Set in Progress         : Set Complete
+Auth Type Support       : MD5
+Auth Type Enable        : Callback : MD5
+                        : User     : MD5
+                        : Operator : MD5
+                        : Admin    : MD5
+                        : OEM      :
+IP Address Source       : Static Address
+IP Address              : 172.16.1.2
+Subnet Mask             : 255.255.254.0
+MAC Address             : f8:bc:12:12:13:14
+SNMP Community String   : public
+IP Header               : TTL=0x40 Flags=0x40 Precedence=0x00 TOS=0x10
+BMC ARP Control         : ARP Responses Enabled, Gratuitous ARP Disabled
+Gratituous ARP Intrvl   : 2.0 seconds
+Default Gateway IP      : 172.16.210.1
+Default Gateway MAC     : 00:00:00:00:00:00
+Backup Gateway IP       : 0.0.0.0
+Backup Gateway MAC      : 00:00:00:00:00:00
+802.1q VLAN ID          : Disabled
+802.1q VLAN Priority    : 0
+RMCP+ Cipher Suites     : 0,1,2,3,4,5,6,7,8,9,10,11,12,13,14
+Cipher Suite Priv Max   : Xaaaaaaaaaaaaaa
+                        :     X=Cipher Suite Unused
+                        :     c=CALLBACK
+                        :     u=USER
+                        :     o=OPERATOR
+                        :     a=ADMIN
+                        :     O=OEM
+Bad Password Threshold  : Not Available
+"""
+
+ipmitool_lan_print_expected = [
+    {"name": "IP Address Source", "value": "Static Address"},
+    {"name": "IP Address", "value": "172.16.1.2"},
+    {"name": "Subnet Mask", "value": "255.255.254.0"},
+    {"name": "MAC Address", "value": "f8:bc:12:12:13:14"},
+    {
+        "name": "BMC ARP Control",
+        "value": "ARP Responses Enabled, Gratuitous ARP Disabled",
+    },
+    {"name": "Default Gateway IP", "value": "172.16.210.1"},
+    {"name": "802.1q VLAN ID", "value": "Disabled"},
+    {"name": "802.1q VLAN Priority", "value": "0"},
+    {"name": "RMCP+ Cipher Suites", "value": "0,1,2,3,4,5,6,7,8,9,10,11,12,13,14"},
+]
+
+SENSOR_DATA_CMD = ["ipmitool", "sdr", "list"]
+IPMITOOL_NORMAL_ERROR = (
+    "Could not open device at /dev/ipmi0 or /dev/ipmi/0 or /dev/ipmidev/0: "
+    "No such file or directory"
+)
+IPMITOOL_ABNORMAL_ERROR = "Unexpected Error!!!"
+
+
+@mock.patch("ipmitool.run_command", autospec=True)
+class TestIpmitool:
+    def test_sensor_normal_error(self, run_command):
```

Comment:
```suggestion
    def test_ipmi_not_available(self, run_command):
```

Reply:
In fact this test is to test the error when running the ipmitool command to get all sensors. There is another test for testing if ipmi is available using the function  `is_ipmi_device_available` that seems corresponds better to test_ipmi_not_available.

Reply:
but agree that `test_sensor_normal_error` is not appropriate. I will change for `test_sensor_data_failed`.

Reply:
or maybe `test_sensor_data_known_failure` because there is a `test_sensor_data_unknown_failure`...

Reply:
And also it could be done for all commands because they can all succeed, failed with a "normal" error and failed with an "abnormal" error. I thought that it will make a lot of tests but it is probably a good thing :)